code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_buff.dm

```diff
@@ -0,0 +1,80 @@
+// A buff given to people sacrificed to help them survive.
+
+/// Screen alert for the below status effect.
+/atom/movable/screen/alert/status_effect/unholy_determination
+	name = "Unholy Determination"
+	desc = "You appear in a unfamiliar room. The darkness begins to close in. Panic begins to set in. There is no time. Fight on, or die!"
+	icon_state = "wounded"
+
+/// The buff given to people within the shadow realm to assist them in surviving.
+/datum/status_effect/unholy_determination
+	id = "unholy_determination"
+	duration = 3 MINUTES // Given a default duration so no one gets to hold onto this buff forever by accident.
+	tick_interval = 1 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/unholy_determination
+	/// How much to heal every second
+	var/heal_per_second = 0.25
+
+/datum/status_effect/unholy_determination/on_creation(mob/living/new_owner, set_duration)
+	if(isnum(set_duration))
+		duration = set_duration
+	return ..()
+
+
+/datum/status_effect/unholy_determination/tick()
+	// The amount we heal of each damage type per tick.
+	var/healing_amount = heal_per_second
+
+	// In softcrit you're, strong enough to stay up.
+	if(owner.health <= 0 && owner.health >= -100)
```

Comment:
I believe we have threshold defines for these

Reply:
Done